subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc

```diff
@@ -1163,6 +1163,23 @@ you should consider configuring such normalization in a <<sharing_build_logic_be
 The effect of this configuration would be that changes to `build-info.properties` would be ignored for up-to-date checks and <<build_cache.adoc#build_cache,build cache>> key calculations.
 Note that this will not change the runtime behavior of the `test` task — i.e. any test is still able to load `build-info.properties` and the runtime classpath is still the same as before.
 
+[[sec:property_file_normalization]]
+==== Properties file normalization
+
+By default, properties files (i.e. files that end in a `.properties` extension) will be normalized to ignore differences in comments, whitespace and the order of properties.  It is sometimes the case, though, that certain properties have a runtime impact, while others do not.  If a property is changing that does not have an impact on the runtime classpath, it may be desirable to exclude it from up-to-date checks and <<build_cache.adoc#build_cache,build cache>> key calculations.  However, excluding the entire file would also exclude the properties that do have a runtime impact.  In this case, properties can be excluded selectively from any or all properties files on the runtime classpath.
+
```

Comment:
I miss here the description of how we handle the file patterns. They are globs, right? Perhaps linking to some definitions like https://en.wikipedia.org/wiki/Glob_(programming) would also be nice.

Reply:
I added a description to `RuntimeClasspathNormalization`.  I don't think it fully matches the description of globbing you linked as it doesn't support things like ranges and the `!` operator.  I think it is based off of the file matching supported by Ant (https://ant.apache.org/manual/dirtasks.html) but it didn't feel right to link to that since it wasn't specific to the file globbing.  I'm open to suggestions if you think this still needs more work.

Reply:
I'm not hostile to the idea of making the globbing more standard.  I think it definitely needs to be a 7.0 change, though, as this uses the same matching capability used in other places like copy/zip tasks, etc.  This could potentially introduce breaking changes to those things.

As far as the differences from git globbing, I'm not sure if you are asking about the basic globbing in https://en.wikipedia.org/wiki/Glob_(programming) or the full git pathspec patterns.  We don't support ranges or negation (i.e. `!` operator) or bracket expressions from the basic globbing.  As far as git pathspecs are concerned, we also don't support special expressions (e.g. `[:digit:]`) or the magic signature stuff (e.g. `:(icase)`). 

My only reservation about this would be that I am not aware of much demand from the community for additional functionality here.  Which is not to imply that there isn't - only that I'm not aware of it.  Although it's probably worth doing, it might be tough to prioritize it without a certain degree of demand.

Reply:
Mhm, maybe this ship has sailed. Anyway, thanks for thinking it through.